.github/workflows/contracts.yml

```
@@ -0,0 +1,200 @@
+        || needs.contracts-detect-changes.outputs.path-filter == 'true'
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./solidity/random-beacon
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "14.x"
+          cache: "yarn"
+          cache-dependency-path: solidity/random-beacon/yarn.lock
+
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.8.5
+
+      - name: Install Solidity
+        env:
+          SOLC_VERSION: 0.8.6 # according to solidity.version in hardhat.config.js
+        run: |
+          pip3 install solc-select
+          solc-select install $SOLC_VERSION
+          solc-select use $SOLC_VERSION
+
+      - name: Install Slither
+        env:
+          SLITHER_VERSION: 0.8.0
+        run: pip3 install slither-analyzer==$SLITHER_VERSION
+
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Run Slither
+        run: slither --hardhat-artifacts-directory build .
+  
+  contracts-build-and-test:
+    needs: contracts-detect-changes
+    if: |
+      github.event_name != 'pull_request'
+        || needs.contracts-detect-changes.outputs.path-filter == 'true'
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./solidity/random-beacon
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "14.x"
+            
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Build solidity contracts
+        run: yarn build
+
+      - name: Run tests
+        run: yarn test
+
+  contracts-deployment-dry-run:
+    if: |
+      github.event_name != 'pull_request'
+        || needs.contracts-detect-changes.outputs.path-filter == 'true'
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./solidity/random-beacon
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "14.x"
+          cache: "yarn"
+          cache-dependency-path: solidity/random-beacon/yarn.lock
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Deploy contracts
+        run: yarn deploy
+
+  contracts-deployment-testnet:
+    needs: [contracts-detect-changes, contracts-build-and-test]
+    if: github.event_name == 'workflow_dispatch'
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./solidity/random-beacon
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Load environment variables
+        uses: keep-network/ci/actions/load-env-variables@v1
+        with:
+          environment: ${{ github.event.inputs.environment }}
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "14.x"
+          cache: "yarn"
+          cache-dependency-path: solidity/random-beacon/yarn.lock
```

```suggestion
          cache-dependency-path: solidity/random-beacon/yarn.lock
          registry-url: "https://registry.npmjs.org"
```

If we're using `NODE_AUTH_TOKEN` env variable in the `Publish to npm` step, we need to specify the `registry-url` property in the step setting up the node.
Otherwise publication of the package will fail with error like this:
```
 '@keep-network/random-beacon@0.0.1-ropsten.0' is not in the npm registry
```

---

I can't see this change here, I think it was added accidentally to `contracts-lint` node setup step